mountinfo/doc.go

```diff
@@ -0,0 +1,41 @@
+// Package mountinfo provides a set of functions to retrieve information about OS mounts.
+// Currently it supports Linux. For historical reasons, there is also some support for FreeBSD,
+// and a shallow implementation for Windows, but in general this is Linux-only package, so
+// the rest of the document only applies to Linux, unless explicitly specified otherwise.
+//
+// In Linux, information about mounts seen by the current process is available from
+// /proc/self/mountinfo. Note that due to mount namespaces, different processes can
+// see different mounts. A per-process mountinfo table is available from /proc/<PID>/mountinfo,
+// where <PID> is a numerical process identifier.
+//
+// In general, /proc is not a very effective interface, and mountinfo is not an exception.
+// For example, there is no way to get information about a specific mount point (i.e. it
+// is all-or-nothing). This package tries to hide the /proc ineffectiveness by using
+// parse filters while reading mountinfo. A filter can skip some entries, or stop
+// processing the rest of the file once the needed information is found.
+//
+// For any functions that have path as an argument (such as Mounted or various filters),
+// the argument must be an absolute path, with all symlinks resolved, and cleaned.
```

Comment:
```suggestion
// the argument must be an absolute path, with all symlinks resolved and cleaned.
```

Reply:
the "clean" part is not about symlinks, but about whatever filepath.Clean() does. If we drop a comma here, it sounds like "cleaned" is about symlinks.

Reply:
Ah, yes, makes sense